test/langtools/tools/javac/doctree/CoverageTest.java

```diff
@@ -0,0 +1,183 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Scans for tests in the same source directory that use the DocCommentTester framework,
+ * and for those tests, scans for doc comments, counting the number of occurrences
+ * of different kinds of DocTree nodes.  The number are mostly just informational,
+ * except for the (implicit) zeroes, indicating that no instances of a kind of tree
+ * have been detected ... which is a hole in the coverage.
+ *
+ * If desired, we could enhance the DocTreeScanner to track different variants of
+ * specific kinds of tree nodes.
+ */
+public class CoverageTest {
+    public static void main(String... args) throws Exception {
+        new CoverageTest().run(args);
+    }
+
+    void run(String... args) throws Exception {
+        Path src = Path.of(System.getProperty("test.src"));
+
+        JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+        StandardJavaFileManager fm = compiler.getStandardFileManager(null, null, null);
+
+        List<JavaFileObject> files = new ArrayList<>();
+        try (DirectoryStream<Path> ds = Files.newDirectoryStream(src, this::isDocCommentTesterTest)) {
+            for (Path p : ds) {
+                files.add(fm.getJavaFileObjects(p).iterator().next());
+            }
+        }
+
+        JavacTask task = (JavacTask) compiler.getTask(null, null, null, null, null, files);
+
+        DocTrees trees = DocTrees.instance(task);
+        Map<DocTree.Kind, Integer> counts = new TreeMap<>();
+        DocTreeScanner dtScanner = new DocTreeScanner() {
+            @Override
+            public Object scan(DocTree node, Object o) {
+                if (node != null) {
+                    DocTree.Kind k = node.getKind();
+                    counts.put(k, counts.computeIfAbsent(k, k_ -> 0) + 1);
+                }
+                return super.scan(node, o);
+            }
+        };
+
+        TreePathScanner declScanner = new DeclScanner() {
+            @Override
+            void visitDecl(Tree tree, Name name) {
+                TreePath path = getCurrentPath();
+                DocCommentTree dc = trees.getDocCommentTree(path);
+                dtScanner.scan(dc, null);
+            }
+        };
+
+
+        task.addTaskListener(new TaskListener() {
+            @Override
+            public void finished(TaskEvent e) {
+                if (e.getKind() == TaskEvent.Kind.PARSE) {
+                    declScanner.scan(e.getCompilationUnit(), null);
+                }
+            }
+        });
+
+        task.parse();
+
+        counts.forEach((k, v) -> System.err.printf("%20s: %5d%n", k, v));
+
+        List<DocTree.Kind> notFound = Stream.of(DocTree.Kind.values())
+                .filter(k -> switch (k) { case DOC_TYPE, OTHER -> false; default -> true; })
```

Comment:
DOC_TYPE only appears in HTML files, and only then processing the entire file (i.e. not just the `<body>` element, as in `package.html` etc. So, it cannot be generated by `.java` files.

OTHER is a special kind that can be used to help provide compatible evolution of visitors/etc. It does not appear in any standard tree nodes.

Reply:
Added this explanation as comments in the file.